nemo/collections/nlp/callbacks/state_tracking_trade_callback.py

```diff
@@ -33,27 +34,39 @@ def eval_iter_callback(tensors, global_vars, data_desc):
     if 'gating_preds' not in global_vars:
         global_vars['gating_preds'] = []
 
-    for kv, v in tensors.items():
-        if kv.startswith('loss'):
-            loss_numpy = v[0].cpu().numpy()
-            global_vars['loss'].append(loss_numpy)
-        if kv.startswith('point_outputs'):
-            point_outputs = v[0]
-        if kv.startswith('gate_outputs'):
-            gate_outputs = v[0]
-        if kv.startswith('gating_labels'):
-            gating_labels = v[0].cpu().numpy()
-            global_vars['gating_labels'].extend(gating_labels)
-        if kv.startswith('tgt_ids'):
-            tgt_ids = v[0]
-
-    point_outputs_max = torch.argmax(point_outputs, dim=-1)
-    mask_paddings = tgt_ids == data_desc.vocab.pad_id
-    comp_res = (point_outputs_max == tgt_ids) | mask_paddings
-    comp_res = torch.all(comp_res, axis=-1, keepdims=False)
-
-    global_vars['comp_res'].extend(comp_res.cpu().numpy())
-    global_vars['gating_preds'].extend(torch.argmax(gate_outputs, axis=-1).cpu().numpy())
+    point_outputs_max_list = []
+    tgt_ids_list = []
+    gate_outputs_max_list = []
+    for tensor_name, values_list in tensors.items():
+        if tensor_name.startswith('loss'):
```

Comment:
you don't use `global_vars[loss]` in the epoch_done(), why do you need it here?

Reply:
It was there for debugging to have the loss for evaluation. I would remove it.